L1Trigger/L1THGCal/plugins/geometries/HGCalTriggerGeometryHexLayerBasedImp1.cc

```diff
@@ -0,0 +1,737 @@
+#include "FWCore/ParameterSet/interface/FileInPath.h"
+
+#include "L1Trigger/L1THGCal/interface/HGCalTriggerGeometryBase.h"
+#include "DataFormats/ForwardDetId/interface/HGCalDetId.h"
+#include "DataFormats/HcalDetId/interface/HcalDetId.h"
+#include "DataFormats/ForwardDetId/interface/ForwardSubdetector.h"
+
+#include <vector>
+#include <iostream>
+#include <fstream>
+#include <regex>
+
+
+class HGCalTriggerGeometryHexLayerBasedImp1 : public HGCalTriggerGeometryBase
```

Comment:
Maybe it would be useful to add a README documenting the different HGCal trigger geometry implementations? It's not immediately clear from the naming what all of the differences are.